program/src/state.rs

```diff
@@ -224,15 +238,21 @@ impl<'a> AddressLookupTable<'a> {
         Ok(data)
     }
 
-    /// Mutably deserialize addresses from a lookup table's data.
+    // [Core BPF]: This is a new function that was not present in the legacy
+    // built-in implementation.
+    /// Mutably deserialize addresses from a lookup table's data. This function
+    /// accepts an index in the list of addresses to start deserializing from.
     pub fn deserialize_addresses_from_index_mut(
         data: &mut [u8],
-        start_index: usize,
+        index: usize,
```

Comment:
nit-up-to-you: since this is now using `index` like the `lookup` function, you could make it slightly clearer by making this a `u8` instead of `usize`

Reply:
I thought about this a bit. My original thought was:

> I'm inclined to leave it as-is, since it feels weird to request a `u8` only to immediately cast to `usize`. The `lookup` function imo should be requesting `&[usize]`, since it maps them all to `usize` anyway.

However, since the index can't be over `255`, this and `lookup` actually make sense as `u8` indices. Changed!